### ISSUE TITLE
typo in config key should break validation

### ISSUE THREAD

**Issue by javipolo** [nmstate/nmstate#1997]
## Describe the bug
When having a wrong key in nmstate yaml, it is ignored when generating configurations, and then a wrong configuration is generated without any kind of warnings or errors

In newer versions, this is not an issue anymore, and it would be great to have that to be backported to 1.3.0

## Nmstate version
- Version 1.3.0

## How reproducible
- 100%

## Steps to Reproduce
```
interfaces:
- name: eth0
  type: ethernet
  state: up
  # Note the typo here, ipv5 instead of ipv4
  ipv5:
    address:
    - ip: 192.168.122.250
      prefix-length: 24
    enabled: true
```
## Nmstate output
- Expected output:
```
$ nmstatectl gc nmstate.yaml
Error: unknown field `ipv5`
```

- Current output:
```
$ nmstatectl gc nmstate.yaml
---
NetworkManager:
- - eth0.nmconnection
  - '[connection]

    id=eth0

    uuid=dd502456-0276-45e5-862d-e4566854a8e8

    type=ethernet

    interface-name=eth0


    [ethernet]


    [ipv4]

    dhcp-client-id=mac

    method=disabled


    [ipv6]

    addr-gen-mode=eui64

    dhcp-duid=ll

    dhcp-iaid=mac

    method=disabled


    [proxy]

    '
```
### Additional context
We're using nmstate as part of openshift assisted-installer, and it's very frustrating to get configurations with typos pass without any type of validation error

**Comment by cathay4t** [comment]
With nmstate 2.1+ (rust rewrite, targeting for RHEL 9.1+), the output changed to:

```
[fge@c9s ~]$ sudo nmstatectl gc /tmp/a.yml
serde_yaml::Error: unknown field `ipv5`
```

The 1.x branch is entering maintenance stage for RHEL 8.x.

@javipolo Do you still need fix in 1.x branch?

**Comment by javipolo** [comment]
Yes @cathay4t, we would need the fix for the 1.x version ... :sweat_smile: 

We're working to migrate to the rust rewrite, but we're still far from it
